mne/stats/cluster_level.py

```diff
@@ -1729,3 +1737,310 @@ def summarize_clusters_stc(
+    buffer_size: int = None,  # block size for chunking the data
+):
+    """
+    Run a cluster permutation test based on formulaic input.
+
+    # currently only supports paired t-test on evokeds or epochs
+
+    Parameters
+    ----------
+    dataframe : pd.DataFrame
+        Dataframe with evoked/epoched data, conditions and subject IDs.
+    formula : str, optional
+        Wilkinson notation formula for design matrix. Default is None.
+    n_permutations : int, optional
+        Number of permutations. Default is 10000.
+    seed : None | int | np.random.RandomState, optional
+        Seed for the random number generator. Default is None.
+    tail : int, optional
+        0 for two-tailed, 1 for greater, -1 for less. Default is 0.
+    n_jobs : int, optional
+        How many cores to use. Default is 1.
+    adjacency : None, optional
+        Adjacency matrix. Default is None.
+    max_step : int, optional
+        Maximum distance between samples (time points). Default is 1.
+    exclude : np.Array, optional
+        Exclude no time points or channels. Default is None.
+    step_down_p : int, optional
+        Step down in jumps test. Default is 0.
+    t_power : int, optional
+        Weigh each location by its stats score. Default is 1.
+    out_type : str, optional
+        Output type. Default is "indices".
+    check_disjoint : bool, optional
+        Check if clusters are disjoint. Default is False.
+    buffer_size : int, optional
+        Block size for chunking the data. Default is None.
+    seed : int, optional
+        Seed for the random number generator. Default is None.
+
+    Returns
+    -------
+    ClusterResult
+        Object containing the results of the cluster permutation test.
+    """
+    # for now this assumes a dataframe with a column for evoked data or epochs
+    # add a data column to the dataframe (numpy array)
+    df["data"] = [evoked.data for evoked in df.evoked]
+
+    # extract number of channels and timepoints
+    # (eventually should also allow for frequency)
+    n_channels, n_timepoints = df["data"][0].shape
+
+    # convert wide format to long format for formulaic
+    df_long = unpack_time_and_channels(df)
+
+    # pivot the DataFrame
+    pivot_df = df_long.pivot_table(
+        index=["subject_index", "channel", "timepoint"],
+        columns="condition",
+        values="value",
+    ).reset_index()
+
+    # if not 2 unique conditions raise error
+    if len(pd.unique(df.condition)) != 2:
+        raise ValueError("Condition list needs to contain 2 unique values")
+
+    # Get unique elements and the indices of their first occurrences
+    unique_elements, indices = np.unique(df.condition, return_index=True)
+
+    # Sort unique elements by the indices of their first occurrences
+    conditions = unique_elements[np.argsort(indices)]
+
+    # print the contrast used for the paired t-test
+    print(f"Contrast used for paired t-test: {conditions[0]} - {conditions[1]}")
```

Comment:
don't use `print()` in production code, use `logger.info`, `logger.warn`, or `logger.debug`. This one should probably be `info` level